src/AppInstallerCommonCore/Public/winget/UserSettings.h

```diff
@@ -200,7 +202,11 @@ namespace AppInstaller::Settings
         SETTINGMAPPING_SPECIALIZATION(Setting::LoggingChannelPreference, std::vector<std::string>, Logging::Channel, Logging::Channel::Defaults, ".logging.channels"sv);
         // Interactivity
         SETTINGMAPPING_SPECIALIZATION(Setting::InteractivityDisable, bool, bool, false, ".interactivity.disable"sv);
-        
+
+        // Experiments
+        using Experiments_t = std::map<std::string, bool>;
+        SETTINGMAPPING_SPECIALIZATION(Setting::Experiments, Experiments_t, Experiments_t, {}, ".experiments"sv);

```

Comment:
This allows to enable/disable individual experiments. As a user, I would like to be able to disable all experimentation. I.e., the same we have for group policy, but available for a single user. If somebody wanted to disable all experiments, they would have to stay on the look for any experiments we add, and that's not good

Reply:
I like this idea. I can add a boolean `enableExperiments` field in user settings. Disabling this would result in disabling all experiments. Thoughts?

Reply:
As long as the default value in the client is set so that experiments are enabled by default (excluding GPO), I'm fine with having a setting a user could add to the client to disable experimentation. The real concern here is that if we're doing experiments, we need to know when a user disables them if an experiment is running so they can be excluded from the results.

Experimentation is only valid when a user doesn't know if they are in the control group or the treatment group. The same is true if a user is receiving an experiment and they disable an experimental feature that was enabled through experimentation.

We would benefit most by having the ability to enable experimental treatments for devices on Windows Insider Canary and Dev builds for features that we're developing (once they are ready) so we can determine if they are working properly, or if users are getting a better desired outcome.

I think we'll need to send a telemetry event if a user opts-out of experimentation and an experiment is running. I also think we'll need to send a telemetry event related to the experimental treatment if a user disables a feature that was enabled via an experiment.

Obviously if a user has disabled telemetry, we wouldn't receive any events, and if an experiment is running and telemetry is disabled, we will no longer get that data, so I'd also say we should disable any experiments if a user disables telemetry.

Reply:
> The real concern here is that if we're doing experiments, we need to know when a user disables them if an experiment is running so they can be excluded from the results.

In the telemetry he's already including whether the experiment is enabled and the reason, i.e. whether it was disabled by GP, overridden from settings, or came from being in the control/treatment group. That should be enough to make a distinction; we just have to add a new reason for the experiment state - "disabled by settings"

I imagine for the comparison we would ignore all groups except people who do not have any override, which should have about the same split in population size as the general population

> I can add a boolean `enableExperiments` field in user settings. Disabling this would result in disabling all experiments. Thoughts?

Sounds good to me, except that I would call it `allow` instead of `enable`